java/arcs/core/common/CounterFlow.kt

```diff
@@ -0,0 +1,33 @@
+package arcs.core.common
+
+import kotlinx.atomicfu.atomic
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+
+/**
+ * Provides a counter with increment/decrement functionality whose value can be observed via a
+ * Flow<Int>.
+ *
+ * The counter value will be conflated: if a large number of changes occur at once, collectors of
+ * the flow may not receive all of them, but are guaranteed to receive the most recent one.
+ */
```

Comment:
Q: (for everyone) what's the difference between arcs/core/common and arcs/core/util?

Reply:
Ohhh, that's a good question...